markets/perps-market/contracts/storage/PerpsMarket.sol

```diff
@@ -96,77 +99,121 @@ library PerpsMarket {
      */
     function maxLiquidatableAmount(
         Data storage self,
-        uint256 requestedLiquidationAmount
+        uint128 requestedLiquidationAmount
     ) internal returns (uint128 liquidatableAmount) {
         PerpsMarketConfiguration.Data storage marketConfig = PerpsMarketConfiguration.load(self.id);
+        uint liquidationIndex = block.timestamp % marketConfig.maxSecondsInLiquidationWindow;
 
         // if endorsedLiquidator is configured and is the sender, allow full liquidation
         if (msg.sender == marketConfig.endorsedLiquidator) {
-            return requestedLiquidationAmount.to128();
-        }
+            self.liquidationAmounts[liquidationIndex] += requestedLiquidationAmount;
```

Comment:
Strictly speaking, it should be 
`self.liquidationAmounts[liquidationIndex] += requestedLiquidationAmount;` if the previous timestamp is in the window, otherwise it should be 
`self.liquidationAmounts[liquidationIndex] = requestedLiquidationAmount;`

Reply:
trueee

resolved